test/e2e/hpa_test.go

```diff
@@ -0,0 +1,96 @@
+package e2e
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/utils/pointer"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+var _ = ginkgo.Describe("hpa replicas synchronization testing", func() {
+	ginkgo.Context("Replicas synchronization testing", func() {
+		var policyNamespace, policyName string
+		var namespace, deploymentName, hpaName string
+		var deployment *appsv1.Deployment
+		var hpa *autoscalingv2.HorizontalPodAutoscaler
+		var policy *policyv1alpha1.PropagationPolicy
+		var retainRIC *configv1alpha1.ResourceInterpreterCustomization
+
+		ginkgo.BeforeEach(func() {
+			policyNamespace = testNamespace
+			namespace = testNamespace
+			policyName = deploymentNamePrefix + rand.String(RandomStrLength)
+			deploymentName = policyName
+			hpaName = policyName
+
+			retainRIC = helper.NewRetainResourceInterpreterCustomization(namespace, policyName)
```

Comment:
I'm so sorry,  i don't understand why we need this `retainRIC` ? according to our proposal, we don't need this object. 🤔

Reply:
You are right, we don't need it.